### DISCUSSION TITLE
Potential Coordination of Semantic-Related Python Projects

### DISCUSSION THREAD

**Discussion by cknoll** [pysemtec/semantic-python-overview#1] [General]
In https://github.com/RDFLib/OWL-RL/issues/35 the following was pointed out by @nicholascar (member of @RDFLib):

> we might be due a whole rethink about OWL-in-Python and coordinating that a bit better with other projects too such as OwlReady2

This seems **very** reasonable to me.

Background: I work with Python since 2008, mostly in engineering (numpy, scipy, sympy, ...) and web development (django) and made the experience that almost all solutions to basic problems are at most one combination of `pip install ... ` and `import ...` away. However when it comes to semantic web technologies the situation seems to be much less satisfactory. E.g., it took me several hours to accept that currently there seems to be no way of parsing OWL in Manchester syntax to owlready2 objects.

My impression is that on a broad scale semantic (web) technology has not yet really taken off – compared to the huge potential it has.

IMHO, the surprisingly thin, scattered and inconsistent Python support for these technologies contributes to this. In comparison e.g. with numerical machine learning (tensorflow, pytorch, scikit-learn, ...) the number and size of semantic-related python projects is quite small. Numerical machine learning as whole topic has benefited significantly from good python support for its methods. I would really appreciate a similar dynamics for semantic technologies.

I think that Python-based software could, should and probably will contribute to the spread of semantics-related competencies and
knowledge, especially among scientists and engineers. And I am convinced that this would be for the benefit of science and education in general.

Establishing some kind of (self-) "coordination" or at least communication between the existing semantic-related python projects seems therefore promising to reduce friction and spawn more activity.

The following things I would consider worthwile:

1. Establishing a mailing list or similar communication platform for people interested in python and semantics (but not project-specific).
2. Gathering the knowledge which kind of problems are already solveable via python and which are not.
3. Raising awareness for a broader audience (and maybe contributions/funding) to improve the feature set and interoperability of the respective tools.

[This repository](https://github.com/cknoll/semantic-python-overview) is my first a attempt to aspect 2. Feel free to open issues or pull requests.

This current issue could be a place for general discussion on this topic.

**Comment by cknoll** [comment]
I am currently in contact (via email) with the author and maintainer of Owlready2. That package IMHO plays a central role in the semantic python world. I would like it to increase the momentum of that project (more docs, more features, more users, more citations, ...). One aspect of this undertaking is to establish an active **Owlready2-repo on github** as this platform by far has the biggest user base and currently Owlready2 [lives on bitbucket](https://bitbucket.org/jibalamy/owlready2/), where very little interaction takes place. E.g. there is no issue-tracker for the official repo.

Every github-repo needs an owner. IMHO, the best option would be the author. However, his github account (@jibalamy) seems to be inactive. Additionally, my current information is that he would prefer to stick with the bitbucket-based workflow for a while (which I think is fine, but should not prevent the creation of an active github-repo). The next best option for repo-ownership would be an organization, e.g. "semantic-python-interest-group" (other suggestions welcome). For a github organization to make any sense there should be some members, say at least two for the very minimum. Obviously, I would be interested. Anyone else? Maybe @blokhin @stigbd?

**Comment by cknoll** [comment]
Note: I just converted the issue to a "github discussion" (new feature).

**Comment by cknoll** [comment]
Note: I transferred the ownership of this repo from its original place https://github.com/cknoll/semantic-python-overview to the newly founded group https://github.com/pysemtec/semantic-python-overview.

Also: the above idea of founding an group of people interested in both python and semantic technology has become real. Now there is *pysemtec*, currently consisting of
- [a mailing list](https://lists.pysemtec.org/postorius/lists/pysemtec.lists.pysemtec.org/),
- a static webpage: <https://pysemtec.org,
- a [![Gitter Chat](https://badges.gitter.im/pysemtec/community.svg)](https://gitter.im/pysemtec/community?utm_source=badge&utm_medium=badge&utm_campaign=pr-badge),
- and a [github organization](https://github.com/pysemtec)).

if you are interested to join or have questions/suggestions do not hesitate to contact us e.g. via mail 
`"ABC@XYZ.org".replace("ABC", "contact").replace("XYZ", "pysemtec")`.

**Reply by majidaldo** [reply]
@cknoll , i echo your experience almost exactly.

but it looks like there is minimal activity. i suggest cutting down the communication channels to just a mailing list. but i think google groups is better.

**Reply by majidaldo** [reply]
the folks at rdflib have the best visibility into the ecosystem. best to tap into that community. (as i see some coordination issues even within rdflib).

**Comment by johnaffolter** [comment]
I love this!  I am working on a few things here and I would love to help out.  We work in Cannabis Retail at Genetica and deal with Labs, COAs, Whole Sale, Retail, Clinical Studies, Retail Studies, Sentiment, Inventory, Budget/ERP, HR data and Seed to Sale for both Retail and Medical situations so it is a fun use case for ontologies and making a standard because it spans so many sectors.  I would love to hear y'alls' thoughts what that might look like and what are the best Python libraries to work across graph to vector and graph to structured and back.
